apps/src/feedbackBlocks.js

```diff
@@ -51,71 +52,65 @@ var FeedbackBlocks = function(options, missingRequiredBlocks, missingRecommended
 
   this.xml = this.generateXMLForBlocks_(blocksToDisplay);
 
-  this.iframeOptions = {
-    app: options.app,
-    options: {
-      readonly: true,
-      locale: studioApp.LOCALE,
-      localeDirection: studioApp.localeDirection(),
-      baseUrl: studioApp.BASE_URL,
-      skinId: options.skin,
-      level: options.level,
-      blocks: this.xml
-    }
-  };
-
   this.div = document.createElement('div');
-  this.div.setAttribute('id', 'feedbackBlocksContainer');
-
-  this.iframe = document.createElement('iframe');
-  this.iframe.setAttribute('id', 'feedbackBlocks');
-  this.iframe.setAttribute('allowtransparency', 'true');
+  this.div.setAttribute('id', 'feedbackBlocks');
 
-  this.div.appendChild(this.iframe);
+  // Will be set by this.render()
+  this.blockSpaceEditor = undefined;
 };
 
 module.exports = FeedbackBlocks;
 
-/**
- * Generate a url (with params) for the iFrame that's going to hold the
- * blocks.
- * TODO(elijah) replace this whole thing with an implementation that
- * doesn't require an iframe, since we can now have multiple blocklys on
- * the same page
- */
-FeedbackBlocks.prototype.readonlyTemplateUrl_ = function () {
-  var params = {
-    app: this.iframeOptions.app,
-    js_locale: this.iframeOptions.options.locale,
-    locale_dir: this.iframeOptions.options.localeDirection
-  };
-
-  // Simple polyfill for $.params
-  var paramString = Object.keys(params).map(function (key) {
-    return key + "=" + encodeURIComponent(params[key]);
-  }).join("&");
-
-  return "/readonly_template?" + paramString;
-};
+FeedbackBlocks.prototype.render = function () {
+  // Only render if this.div exists in the DOM
+  var div = document.getElementById("feedbackBlocks");
+  if (div !== this.div) {
+    return;
+  }
 
-FeedbackBlocks.prototype.show = function() {
-  var iframeOptions = this.iframeOptions;
-  var iframe = this.iframe;
-  iframe.setAttribute('src', this.readonlyTemplateUrl_());
-  iframe.onload = function () {
-    this[iframeOptions.app + "Main"](iframeOptions.options);
-  }.bind(iframe.contentWindow);
+  // This is sort of a nasty hack. Throughout the blockly code,
+  // constructors look at Blockly.readOnly to determine if they're
+  // supposed to be readonly. Because we want this BlockSpace to be
+  // readonly regardless of the global setting, we temporarily override
+  // the value while we build what we want, then restore it at the end.
+  // Similarly for languageTree, which is supposed to define the toolbox
+  var readOnly = Blockly.readOnly;
+  var languageTree = Blockly.languageTree;
+
+  Blockly.readOnly = true;
+  Blockly.languageTree = false;
```

Comment:
According to [this line](https://github.com/code-dot-org/code-dot-org/blob/staging/blockly-core/core/initialization/inject.js#L105), looks like it should actually be `null`. I'll fix that.